### PR TITLE
CSRA-640: Add REST tests for the prisoner assessment API

### PR DESCRIPTION
**Pull request by retroredge**

### CODE REVIEW

File: test/end-to-end/rest/assessments.spec.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+const baseUrl = process.env.APP_BASE_URL;
+const timeoutDuration = 25000;
+
+function generateNomisId() {
+  const nomisId = uuid().substring(0, 8);
+  return nomisId.toUpperCase();
+}
+
+function primeDatabase(nomisId, riskAssessment, healthAssessment, outcome) {
+  return db
```
Comment (romidane) [inline test/end-to-end/rest/assessments.spec.js]: I think that we can remove the then in this promise?

File: test/end-to-end/rest/assessments.spec.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    git_date: new Date(),
+    questions_hash: 'hash',
+    git_version: 'git_version',
+    risk_assessment: riskAssessment,
+    health_assessment: healthAssessment,
+    outcome,
+  })
+  .into('prisoner_assessments')
+  .returning('id')
+  .then(result => ({ id: result[0] }));
```
Comment (romidane) [inline test/end-to-end/rest/assessments.spec.js]: I think that we can remove the then in this promise?

File: test/end-to-end/utils/dbAssertions.js [hunk 3]
```diff
@@ -50,32 +50,31 @@ const checkThatHealthAssessmentDataWasWrittenToDatabase = ({
...
   forename,
   surname,
   dateOfBirth,
-}) => db.select()
-    .table('prisoner_assessments')
-    .where('id', Number(id))
-    .then((result) => {
-      expect(result[0]).to.not.equal(
+}) {
+  return db.select().table('prisoner_assessments').where('id', Number(id)).then((result) => {
```
Comment (romidane) [inline test/end-to-end/utils/dbAssertions.js]: Curious whether you could rewrite this method using async await by removing the then part?

File: test/end-to-end/utils/dbAssertions.js [hunk 4]
```diff
@@ -1,110 +1,62 @@
...
+export async function checkThatRiskAssessmentDataWasWrittenToDatabase({ id, riskAssessment }) {
+  const result = await db.select()
+  .table('prisoner_assessments')
+  .where('id', Number(id));
 
-const checkThatHealthAssessmentDataWasWrittenToDatabase = ({
-  id,
-  healthAssessment,
-}) => db.select()
+  expect(result[0]).to.not.equal(
```
Comment (romidane) [inline test/end-to-end/utils/dbAssertions.js]: can you store `result[0]` in a variable?

File: test/end-to-end/utils/dbAssertions.js [hunk 5]
```diff
@@ -1,110 +1,62 @@
...
+    undefined,
+    `Did not get a result from database for id: ${id}`,
+  );
+  expect(JSON.parse(result[0].risk_assessment).viperScore).to.eql(riskAssessment.viperScore);
```
Comment (romidane) [inline test/end-to-end/utils/dbAssertions.js]: can we store `JSON.parse(result[0].risk_assessment)` in a variable ?

File: test/end-to-end/utils/dbAssertions.js [hunk 6]
```diff
@@ -1,110 +1,62 @@
...
-
-  Future.wait(future);
-
-  future.get();
-};
+  expect(result[0]).to.not.equal(
+    undefined,
+    `Did not get a result from database for id: ${id}`,
+  );
+  expect(result[0].nomis_id).to.equalIgnoreCase(nomisId);
```
Comment (romidane) [inline test/end-to-end/utils/dbAssertions.js]: can we store `result[0]` in a variable?

File: test/end-to-end/utils/dbClear.js [hunk 7]
```diff
@@ -1,15 +1,7 @@
...
-
-export const clearPrisonerAssessmentsSync = () => {
-  const future = Future.fromPromise(clearPrisonerAssessments());
-
-  Future.wait(future);
-
-  future.get();
-};
+async function clearPrisonerAssessments() {
+  await db.delete().table('prisoner_assessments');
```
Comment (romidane) [inline test/end-to-end/utils/dbClear.js]: This one doesn't need the await syntax

File: test/end-to-end/utils/dbAssertions.js (hunk 4 above)
Comment (retroredge) [reply test/end-to-end/utils/dbAssertions.js]: sure. will do

File: test/end-to-end/utils/dbAssertions.js (hunk 5 above)
Comment (retroredge) [reply test/end-to-end/utils/dbAssertions.js]: Changed it to be like the heath assessment check which does the entire check in one go

File: test/end-to-end/utils/dbAssertions.js (hunk 6 above)
Comment (retroredge) [reply test/end-to-end/utils/dbAssertions.js]: Done

File: test/end-to-end/utils/dbClear.js (hunk 7 above)
Comment (retroredge) [reply test/end-to-end/utils/dbClear.js]: good point. cheers

Comment (romidane) [reply test/end-to-end/utils/dbClear.js]: no problem

File: test/end-to-end/rest/assessments.spec.js [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+      dateOfBirth: 'Sat Dec 31 1988 00:00:00 GMT+0000 (GMT)',
+    });
+  });
+
+  it('retrieves the prisoner assessments list', async function test() {
+    await primeDatabase({ nomisId });
+
+    const result = await request(baseUrl).get('/api/assessments/').expect(200);
+
+    expect(result.body[0].nomisId).to.equal(nomisId);
```
Comment (romidane) [inline test/end-to-end/rest/assessments.spec.js]: Sorry this here could do with the same love

**Pull request merged**